src/lib/components/engineUI/EngineSeperate.svelte

```diff
@@ -0,0 +1,391 @@
+<script lang="ts">
+	import Modal from "../modal/Modal.svelte";
+	import type { EngineDTO } from "./EngineDTO";
+	import {
+		Delete,
+		Close,
+		Done,
+		Check_box,
+		Check_box_outline_blank,
+	} from "svelte-google-materialdesign-icons";
+	import type iModalComponent from "$lib/interfaces/IModalComponent";
+	import {
+		comparePortRange,
+		validateEndPort,
+		validateIP,
+		validateName,
+		validateStartPort,
+	} from "$lib/classes/engine/Validation";
+	import type EngineSeperate from "./EngineSeperate.svelte";
+	import SvgButton from "../buttons/SvgButton.svelte";
+
+	export let currentComponent: EngineSeperate | undefined;
+
+	let modalContainer: Modal & iModalComponent;
+	let nameContainer: HTMLInputElement;
+	let ipAddressContainer: HTMLInputElement;
+	let startPortContainer: HTMLInputElement;
+	let endPortContainer: HTMLInputElement;
+
+	let nameBorderColour: string = "var(--engine-ui-underline-color)";
+	let ipBorderColour: string = "var(--engine-ui-underline-color)";
+	let portColour: string = "var(--engine-ui-underline-color)";
+
+	let engineUIErrorUnderlineColour: string =
+		"var(--engine-ui-error-underline-color)";
+	let engineUIUnderlineColour: string = "var(--engine-ui-underline-color)";
+
+	export let currentEngine: EngineDTO;
+	export let tempEngines: Array<EngineDTO>;
+
+	export const setUpEngineSeperate = () => {
+		changeNameBorder();
+		changeIpBorder();
+		validatePort();
+	};
+
+	/**
+	 * Delete an engine by setting the adress of the engine to -1 and removing it from the view
+	 */
+	function deleteEngine() {
+		currentEngine.address = "-1";
+		if (tempEngines.length == 1) {
+			nameContainer.value = "";
+			ipAddressContainer.value = "";
+			startPortContainer.value = "";
+			endPortContainer.value = "";
+			currentEngine.name = "";
+			closeModal();
+			return;
+		}
+		currentComponent?.$destroy();
```

Comment:
It should not be necessary to use JS calls for this. If you implemented the components to react when you change the array, Svelte will automatically destroy this for you.